### PR TITLE
Fix search by order number in Navigator

### PR DESCRIPTION
**Pull request by Cloud11PL**
## What type of PR is this?
- [ ] 💅 Refactor
- [ ] 🌟 Feature
- [ ] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes #

## Usage Instructions, Screenshots, Recordings

CleanShot 2024-07-23 at 12 15 20

## Have you written tests?
- [ ] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

### CODE REVIEW

File: src/components/NavigatorSearch/modes/orders.ts [hunk 1]
```diff
@@ -17,31 +17,30 @@ export function getGqlOrderId(orderNumber: string): string {
...
-        type: "action",
-      },
-    ];
+  if (!orders) {
+    return [];
   }
 
-  return [];
+  return orders.map(order => {
+    const orderNumber = maybe(() => order.number, "");
```
Comment (poulch) [inline src/components/NavigatorSearch/modes/orders.ts:29]: suggestion: avoid to use maybe
```suggestion
    const orderNumber = order?.number ?? ""
```

File: src/components/NavigatorSearch/modes/orders.ts [hunk 2]
```diff
@@ -17,31 +17,30 @@ export function getGqlOrderId(orderNumber: string): string {
...
+      label: intl.formatMessage(messages.goToOrder, {
+        orderNumber,
+      }),
+      onClick: () => {
+        navigate(orderUrl(order.id));
+
+        return false;
+      },
+      type: "action",
+    };
```
Comment (poulch) [inline src/components/NavigatorSearch/modes/orders.ts:42]: suggestion: you could extract this functio, name it like `toAction`, a use like this
```
return orders.map(toAction)
```

File: src/components/NavigatorSearch/queries/queries.ts [hunk 3]
```diff
@@ -1,10 +1,15 @@
 import { gql } from "@apollo/client";
 
-export const checkIfOrderExists = gql`
-  query CheckIfOrderExists($id: ID!) {
-    order(id: $id) {
-      id
-      status
+export const searchOrdersByNumber = gql`
+  query SearchOrdersByNumber($first: Int!, $query: [String!]) {
```
Comment (poulch) [inline src/components/NavigatorSearch/queries/queries.ts:4]: suggestion: `$query` suggest that variable contains words, maybe name it like this `orderIds`

**Review by poulch — DISMISSED** [review]

File: src/components/NavigatorSearch/queries/queries.ts (hunk 3 above)
Comment (Cloud11PL) [reply src/components/NavigatorSearch/queries/queries.ts:4]: It's technically a search and all modes in Navigator use `query` so for the sake of consistency I'm keeping `query`

**Review by poulch — DISMISSED** [review]

File: src/components/NavigatorSearch/queries/useQuickOrderSearch.test.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+}));
+jest.mock("@dashboard/hooks/useDebounce", () => (fn: any) => fn);
+
+describe("useQuickOrderSearch", () => {
+  it("invokes search query", () => {
+    // Arrange
+    const { result } = renderHook(() => useQuickOrderSearch());
+
+    // Act
+    act(() => result.current[1]("1234"));
```
Comment (poulch) [inline src/components/NavigatorSearch/queries/useQuickOrderSearch.test.ts:19]:
```suggestion
    const { result } = renderHook(() => useQuickOrderSearch());
    const [_, setQueryDebounced] = result.current

    // Act
    act(() => setQueryDebounced("1234"));
```

**Review by poulch — APPROVED** [review]

**Review by poulch — APPROVED** [review]

File: src/components/NavigatorSearch/modes/orders.ts [hunk 5]
```diff
@@ -17,31 +17,30 @@ export function getGqlOrderId(orderNumber: string): string {
...
+  return orders.map(order => {
+    const orderNumber = order?.number ?? "";
+
+    return {
+      extraInfo: transformOrderStatus(order.status, intl).localized,
+      label: intl.formatMessage(messages.goToOrder, {
+        orderNumber,
+      }),
+      onClick: () => {
+        navigate(orderUrl(order.id));
```
Comment (andrzejewsky) [inline src/components/NavigatorSearch/modes/orders.ts:37]: can't entries be a links? navigation based on onClick are hacky a bit

Comment (Cloud11PL) [reply src/components/NavigatorSearch/modes/orders.ts:37]: I can improve it in the scope of the general Navigator improvements ticket (it's in backlog)

**Pull request merged**